mutiny-core/src/nodemanager.rs

```diff
@@ -522,45 +516,51 @@ impl NodeManager {
             }
         });
 
+        // if we found a tx we should try to import it into the wallet
+        if let Some(details) = details_opt.clone() {
+            let mut wallet = self.wallet.wallet.try_write()?;
+
+            wallet
+                .insert_tx(
+                    details.transaction.clone().unwrap(),
+                    details.confirmation_time,
+                )
+                .map_err(|_| MutinyError::ChainAccessFailed)?; // TODO better error
```

Comment:
Is there any problem with this being written multiple times when it is not needed? Perhaps this is being called into it confirms, in which case this `insert_tx` will keep being triggered.

Reply:
The docs for this function make it sound like it is idempotent. It shouldn't be a big issue either as this doesnt write to disk immediately, just stages it as well (according to the docs)